### PR TITLE
feat: add possibility of getting latest track versions from fragmenter distribution manifest

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

 - Add possibility of getting latest track versions from fragmenter distribution manifest
 - Clean up install state detection logic
 - Clean up update checking logic
 - Add developer setting for forcing use of local configuration
 
## Testing Instructions
- Make sure addon installing, updating, version switching, removal, etc. works correctly
- Make sure you cannot get into a broken state by cancelling the install/update/version switch
- Make sure those operations are correctly blocked when SImBridge or the sim are running

### CODE REVIEW

File: src/renderer/components/AddonSection/index.tsx [hunk 1]
```diff
@@ -254,11 +176,10 @@ export const AddonSection = (): JSX.Element => {
   }, [dispatch, publisherData, selectedAddon]);
 
   useEffect(() => {
-    findInstalledTrack();
     if (!isInstalling) {
-      InstallManager.determineAddonInstallState(selectedAddon).then(setCurrentInstallStatus);
+      void InstallManager.getAddonInstallState(selectedAddon);
     }
-  }, [findInstalledTrack, isInstalling, selectedAddon, setCurrentInstallStatus]);
+  }, [isInstalling, selectedAddon]);
```
Comment (FoxtrotSierra6829) [inline src/renderer/components/AddonSection/index.tsx:182]: Is this function still supposed to refresh the state when changing the addon, like it did previously? It doesn't anymore if a state exists (which basically has to based on App\index.tsx line 36-39)

File: src/renderer/utils/InstallManager.tsx [hunk 2]
```diff
@@ -688,6 +594,167 @@ export class InstallManager {
...
+    for (const track of addon.tracks) {
+      const updateInfo = await fragmenterUpdateChecker.needsUpdate(track.url, installDir, { forceCacheBust: true });
+
+      let info: ReleaseInfo;
+      if (track.releaseModel.type === 'fragmenter') {
+        info = {
+          name: updateInfo.distributionManifest.version,
+          changelogUrl: undefined,
+          releaseDate: Date.now(),
+        };
```
Comment (FoxtrotSierra6829) [inline src/renderer/utils/InstallManager.tsx:733]: Can the missing properties be checked for existence instead and included in the fragmenter distributionManifest wherever possible?

**Review by FoxtrotSierra6829 — CHANGES_REQUESTED** [review]

File: src/renderer/components/AddonSection/index.tsx [hunk 3]
```diff
@@ -287,13 +208,16 @@ export const AddonSection = (): JSX.Element => {
...
             }}
             dontShowAgainSettingName="mainSettings.disableExperimentalWarning"
           />,
         );
       } else {
-        selectAndSetTrack(track.key);
+        setCurrentlySelectedTrack(track);
+
+        // Update install state
+        void InstallManager.refreshAddonInstallState(selectedAddon);
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/renderer/components/AddonSection/index.tsx]: Fixed.

File: src/renderer/utils/InstallManager.tsx [hunk 4]
```diff
@@ -688,6 +594,167 @@ export class InstallManager {
...
+      return null;
+    }
+
+    const matchingTrack = addon.tracks.find((it) => it.url === install.source);
+
+    if (!matchingTrack) {
+      return null;
+    }
+
+    this.setCurrentlyInstalledTrack(addon, matchingTrack);
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/renderer/utils/InstallManager.tsx:653]: Both fixed.

File: src/renderer/utils/InstallManager.tsx [hunk 5]
```diff
@@ -688,6 +594,167 @@ export class InstallManager {
...
+    if (Directories.isGitInstall(installDir)) {
+      return { status: InstallStatus.GitInstall };
+    }
+
+    const addonInstalledTrack = this.determineAddonInstalledTrack(addon);
+    const addonSelectedTrack = this.getAddonSelectedTrack(addon);
+
+    if (!addonInstalledTrack) {
+      return { status: InstallStatus.NotInstalled };
+    }
```
[the remark this answers is not in the document]
Comment (Benjozork) [reply src/renderer/utils/InstallManager.tsx]: Bot hfixed.

File: src/renderer/components/AddonSection/index.tsx (hunk 1 above)
Comment (Benjozork) [reply src/renderer/components/AddonSection/index.tsx]: Fixed.

File: src/renderer/utils/InstallManager.tsx (hunk 2 above)
Comment (Benjozork) [reply src/renderer/utils/InstallManager.tsx:757]: The properties currently never exist on the distribution manifest.

File: src/renderer/utils/InstallManager.tsx [hunk 6]
```diff
@@ -688,11 +600,192 @@ export class InstallManager {
...
+
+    if (status) {
+      return status;
+    }
+
+    return this.refreshAddonInstallState(addon);
+  }
+
+  public static async refreshAddonInstallState(addon: Addon): Promise<InstallState> {
+    const currenState = store.getState().installStatus[addon.key] as InstallState;
```
Comment (FoxtrotSierra6829) [inline src/renderer/utils/InstallManager.tsx:622]:
```suggestion
    const currentState = store.getState().installStatus[addon.key] as InstallState;
```
typo :D (don't commit this suggestion)

Comment (Benjozork) [reply src/renderer/utils/InstallManager.tsx]: Fixed.

File: package.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "fbw-installer",
   "productName": "FlyByWire Installer",
-  "version": "3.4.0-dev.1",
```
Comment (Benjozork) [inline package.json:4]: a

File: src/renderer/utils/InstallManager.tsx (hunk 2 above)
Comment (FoxtrotSierra6829) [reply src/renderer/utils/InstallManager.tsx:757]: Can we make them existent? Shouldn't be too hard.

File: src/renderer/utils/InstallManager.tsx [hunk 8]
```diff
@@ -688,11 +600,192 @@ export class InstallManager {
...
+      }
+
+      return { status: InstallStatus.UpToDate };
+    } catch (e) {
+      console.error(e);
+      return { status: InstallStatus.Unknown };
+    }
+  }
+
+  public static async checkForUpdates(addon: Addon): Promise<void> {
```
Comment (FoxtrotSierra6829) [inline src/renderer/utils/InstallManager.tsx:739]: This function needs to check if the installStatus is not one of the installing/updating states before further execution due to it being called periodically in App/index.tsx

**Review by FoxtrotSierra6829 — CHANGES_REQUESTED** [review]

File: src/renderer/utils/InstallManager.tsx (hunk 8 above)
Comment (Benjozork) [reply src/renderer/utils/InstallManager.tsx:739]: Fixed.

**Review by FoxtrotSierra6829 — APPROVED** [review]

**Review by FoxtrotSierra6829 — APPROVED** [review]

**Comment by Saschl** [comment]
Quality Assurance Tester/Dev Team Report

Discord Username  : saschl 
Object of testing : [477 ](https://github.com/flybywiresim/installer/pull/477)
Date              : 08/10/2024 

Testing Process: 
 
Installed A32NX, switched versions. Seems to work properly. Simbridge dependency is being installed if it's not installed yet. 

Installed Simbridge, started tried uninstalling. Popup appears and after stopping installation is successful. 
    -> Note: Simbridge does not seem to shutdown properly when the page on the installer is open. I believe this is already an issue on master
After simbridge is stopped, uninstall works properly.

Cancelled a few installs, state seems to be consistent

Progress for simbridge (when installed as dependency) is being shown on the aircraft page. If that's expected then pass

Testing Results: 
Passed/Not Passed  


Negatives:       

It would be good to show an error if uninstall fails due to certain issues (i.e. file locked/busy which currently happens when simbridge is stopped via the installer, but the process is still running). Right now I only see an error in the console.
Can be a separate PR imo.

Conclusions: 


Media:

**Pull request merged**